frontend/src/stores/levels/actions.ts

```diff
@@ -0,0 +1,47 @@
+import { get, post, put, deleteRequest } from '@/services/http'
+import { levels } from './state'
```

Comment:
You don't need this state. You have to set your levels to the role you are passing

1. Get the levels in the backend, of a specific role
2. Put those levels in the role object, in the role state. For this, you'll need to import the role state here in this file